.gitlab-ci.yml

```
@@ -66,9 +66,24 @@ stages:
   - prerequisites
   - build-and-test
 
+# Template for jobs that should NOT run during performance schedules
+.not-performance-schedule-rule:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule" && $SCHEDULE_TYPE == "weekly_performance"
+      when: never
+    - when: always
+
+# Template for jobs that should run ONLY during performance schedules or manually
+.performance-schedule-rule:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "schedule" && $SCHEDULE_TYPE == "weekly_performance"
+      when: always
+    - when: always
```

Should the default be "do not run" ?
```suggestion
    - when: never
```

---

I was thinking to make the performance test pipeline run on a schedule on main as well as run on new commits on main and new (non-draft) PRs like we do for the regression tests. And then in GitHub it would add the results to the dashboard if they came from main but not if they came from other branches. It should give an alert (both a failing pipeline and a commit comment) on GitHub when a PR has a result that exceeds the threshold. (Sorry for not adding this to the PR description yet)

So I think I want the schedule rule in addition to the default rule we use for the regression tests which I think is `when: always`